### PR TITLE
do not send body with delete collector/deviceid restapi; add must statement for ipaddress/type validation

### PR DESCRIPTION
**Pull request by spenugondaa**
The PR has three changes.
One was to remove body while calling delete() for no collector/deviceid commands.
```
sonic(config-tam)# do show tam collector
------------------------------------------------------------------------------------------------
NAME           IP TYPE        IP ADDRESS     PORT
------------------------------------------------------------------------------------------------
c1             ipv4           1.1.1.1        10
c99            ipv4           1.1.1.1        990
cnew           ipv4           1.1.1.4        40
sonic(config-tam)# no collector cnew
sonic(config-tam)# do show tam collector
------------------------------------------------------------------------------------------------
NAME           IP TYPE        IP ADDRESS     PORT
------------------------------------------------------------------------------------------------
c1             ipv4           1.1.1.1        10
c99            ipv4           1.1.1.1        990
sonic(config-tam)#
sonic(config-tam)# do show tam device
---------------------------------------------------------
TAM Device Information
---------------------------------------------------------
device-id: 100
sonic(config-tam)# no device-id
sonic(config-tam)# do show tam device
---------------------------------------------------------
TAM Device Information
---------------------------------------------------------
device-id: 0


```

The second one is to enforce matching address/type pairs with tam collector. A must statement is added to enforce that user could only enter ipv6 address (with :: delimiter) when type is selected as ipv6 and ipv4 address with type as ipv4.

```
sonic(config-tam)# collector c0072 ip-type ipv4 ip-addr 1::2 port 10
%Error: Wrong ipaddress for the specified ipaddress-type 
sonic(config-tam)# collector c0072 ip-type ipv6 ip-addr 1.1.1.2 port 10
%Error: Wrong ipaddress for the specified ipaddress-type
sonic(config-tam)# collector c0072 ip-type ipv6 ip-addr 1::2 port 10
sonic(config-tam)#

```

The third is to change the parameter name for collector command to be "type" instead of "ip-type"

### CODE REVIEW

File: src/CLI/clitree/cli-xml/tam.xml [hunk 1]
```diff
@@ -92,7 +92,7 @@ limitations under the License.
              help="Collector name" 
              ptype="STRING"> 
       </PARAM>
-      <PARAM name="ip-type" 
+      <PARAM name="type"
```
Comment (asha-behera) [inline src/CLI/clitree/cli-xml/tam.xml:95]: ip-addr also needs to changed to "ip" , in line 104, As the correct comamnd is "collector <collector-name> type <ipv4/ipv6> ip <ipaddress> port <collector-port>"

File: models/yang/sonic/sonic-tam.yang [hunk 2]
```diff
@@ -52,6 +52,12 @@ module sonic-tam {
                 }
 
                 leaf ipaddress-type {
+                    must "(contains(../ipaddress, ':') and current()='ipv6') or
+                          (contains(../ipaddress, '.') and current()='ipv4')" {
+                        error-message "%Error: Wrong ipaddress for the specified ipaddress-type";
+                        error-app-tag "%Error: Wrong ipaddress for the specified ipaddress-type";
```
Comment (dutta-partha) [inline models/yang/sonic/sonic-tam.yang]: This is incorrect error-app-tag format . Please refer to https://tools.ietf.org/html/rfc7950. 
Should be like :
error-app-tag invalid-ipaddress.

File: src/CLI/clitree/cli-xml/tam.xml (hunk 1 above)
Comment (spenugondaa) [reply src/CLI/clitree/cli-xml/tam.xml:95]: Done

File: models/yang/sonic/sonic-tam.yang (hunk 2 above)
Comment (spenugondaa) [reply models/yang/sonic/sonic-tam.yang]: Done

**Review by asha-behera — APPROVED** [review]

**Pull request merged**